### PR TITLE
Move DocumentOrShadowRoot.getSelection to Document.getSelection

### PR DESCRIPTION
**Pull request by Elchi3**
This PR is the gold example for why the mixin work matters.
The current page is this: https://developer.mozilla.org/en-US/docs/Web/API/DocumentOrShadowRoot/getSelection

There are so many things wrong here:
- This API is an old school Document API. It is supported since forever.
- The spec defines it just for Document. https://w3c.github.io/selection-api/#extensions-to-document-interface
- No spec defines it under DocumentOrShadowRoot. This is not under any mixin!
- It is surely not experimental
- It has nothing to do with Shadow DOM!

So basically, we warn about a perfect API that's been around forever as experimental and we document it weirdly under DocumentOrShadowRoot. Such a shame :(

I therefore moved it back to Document, cleaned up the page, and updated links to it.

### CODE REVIEW

File: files/en-us/web/api/document/selectionchange_event/index.html [hunk 1]
```diff
@@ -74,6 +74,6 @@ <h2 id="See_also">See also</h2>
 
 <ul>
  <li>{{domxref("Document/selectstart_event", "selectstart")}}</li>
- <li>{{domxref("DocumentOrShadowRoot/getSelection", "Document.getSelection")}}</li>
```
Comment (chrisdavidmills) [inline files/en-us/web/api/document/selectionchange_event/index.html:77]: Nitpick — this is a method, so should be written with parens after it.

**Review by chrisdavidmills — CHANGES_REQUESTED** [review]
This one is funny; it must have existed on the mixin at some point for us to have documented it like this, but I've tested it, and sure enough, it does not exist on ShadowRoot. Also looking at [searchfox](https://searchfox.org/mozilla-central/source/dom/webidl/DocumentOrShadowRoot.webidl),  it does not exist on the mixin either. So I don't think we can ever trust mixins for documentation really. Those implementation details are too subject to change and inconsistent across browsers.

Anyway, nice work @Elchi3 ; I've just got one little nitpick for you to look at, then we can merge.

**Comment by Elchi3** [comment]
Thanks Chris! I've added the ().

Philip found out the reason in https://github.com/mdn/browser-compat-data/pull/9431. Chromium exposes this to ShadowRoot as a non-standard feature. It is unclear if that behavior is standardized or removed in the future. Nevertheless, this PR makes the docs for Document.getSelection clear and doesn't confuse everything with a potential (and currently non-standard) Shadow DOM feature.

**Comment by Elchi3** [comment]
To be clear, if it ever becomes a standard, the correct thing is to add docs to ShadowRoot directly at that point.
I'm not intending to do that now.

**Review by chrisdavidmills — APPROVED** [review]
Looks perfect, thanks @Elchi3 !

And thanks for the explanation about where this weirdness came from. I did wonder, and it is interesting how these messes come about.
